generate.go

```diff
@@ -121,27 +122,57 @@ func cloneTemplate(pdk pdkTemplate, dir, tag string) error {
 		return err
 	}
 
-	if err := runCmdInDir(dir, "git", "checkout", "--orphan", "extism-init", "main"); err != nil {
+	// recursively check that parents are not a git repository, create an orphan branch & commit, cleanup
+	// otherwise, remove the git repository and assume this should be a plain directory within the parent
+	absDir, err := filepath.Abs(dir)
+	if err != nil {
 		return err
 	}
+	if hasGitRepoInParents(absDir, 100) {
+		if err := os.RemoveAll(filepath.Join(dir, ".git")); err != nil {
+			return err
+		}
+	} else {
+		if err := runCmdInDir(dir, "git", "checkout", "--orphan", "extism-init", "main"); err != nil {
+			return err
+		}
 
-	if err := runCmdInDir(dir, "git", "commit", "-am", "init: extism"); err != nil {
-		return err
-	}
+		if err := runCmdInDir(dir, "git", "commit", "-am", "init: extism"); err != nil {
+			return err
+		}
 
-	if err := runCmdInDir(dir, "git", "branch", "-M", "extism-init", "main"); err != nil {
-		return err
-	}
+		if err := runCmdInDir(dir, "git", "branch", "-M", "extism-init", "main"); err != nil {
+			return err
+		}
 
-	if err := runCmdInDir(dir, "git", "remote", "remove", "origin"); err != nil {
-		return err
+		if err := runCmdInDir(dir, "git", "remote", "remove", "origin"); err != nil {
+			return err
+		}
 	}
 
 	fmt.Println("Generated", pdk.Name, "plugin scaffold at", dir)
 
 	return nil
 }
 
+func hasGitRepoInParents(dir string, depth int) bool {
+	parent := filepath.Dir(dir)
+	if depth == 0 || parent == "" || parent == "." || parent == dir {
```

Comment:
i think if parent `parent == "/"` we probably also want to return false here

Reply:
Likely pedantic, but is it possible there is a .git repo there?

Reply:
oh yeah maybe in docker

Reply:
hah I really don't know, so if you think we should return false here too I'm ok with it -- was just thinking it could be possible although pretty unrealistic?

Reply:
no i think you're right - i was missing that `parent == dir` ensures that the recursion will stop at `/`